DocsCode/NavigationTutorial/NavigationTutorialViewController.swift

```diff
@@ -0,0 +1,122 @@
+// #-code-snippet: navigation dependencies-swift
+import Mapbox
+import MapboxCoreNavigation
+import MapboxNavigation
+import MapboxDirections
+// #-end-code-snippet: navigation dependencies-swift
+
+class ViewController: UIViewController, MGLMapViewDelegate {
+    var mapView: MGLMapView!
+    // #-code-snippet: navigation directions-route-swift
+    var directionsRoute: Route?
+    // #-end-code-snippet: navigation directions-route-swift
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // #-code-snippet: navigation init-map-swift
+        mapView = NavigationMapView(frame: view.bounds)
+        mapView.setCenter(CLLocationCoordinate2D(latitude: 30.265, longitude: -97.741), zoomLevel: 11, animated: false)
+        view.addSubview(mapView)
+        // Set the map view's delegate
+        mapView.delegate = self
+        // #-end-code-snippet: navigation init-map-swift
+        
+        // #-code-snippet: navigation user-location-swift
+        // Allow the map view to display the user's location
+        mapView.showsUserLocation = true
+        // #-end-code-snippet: navigation user-location-swift
+        
+        // #-code-snippet: navigation gesture-recognizer-swift
+        // Add a gesture recognizer to the map view
+        let setDestination = UILongPressGestureRecognizer(target: self, action: #selector(didLongPress(_:)))
+        mapView.addGestureRecognizer(setDestination)
+        // #-end-code-snippet: navigation gesture-recognizer-swift
+    }
+    
+    // #-code-snippet: navigation allow-callouts-swift
+    // Implement the delegate method that allows annotations to show callouts when tapped
+    func mapView(_ mapView: MGLMapView, annotationCanShowCallout annotation: MGLAnnotation) -> Bool {
+        return true
+    }
+    // #-end-code-snippet: navigation allow-callouts-swift
+    
+    // #-code-snippet: navigation long-press-swift
+    @objc func didLongPress(_ sender: UILongPressGestureRecognizer) {
+        guard sender.state == .began else { return }
+        
+        // Converts point where user did a long press to map coordinates
+        let point = sender.location(in: mapView)
+        let coordinate = mapView.convert(point, toCoordinateFrom: mapView)
+        
+        // Create a basic point annotation and add it to the map
+        let annotation = MGLPointAnnotation()
+        annotation.coordinate = coordinate
+        annotation.title = "Start navigation"
+        mapView.addAnnotation(annotation)
+        
+        // Calcuate the route from the user's location to the set destination
+        calculateRoute(from: (mapView.userLocation!.coordinate), to: annotation.coordinate) { [unowned self] (route, error) in
```

Comment:
There’s no use of `self` in this callback, so `[unowned self]` should be unnecessary.